### DISCUSSION TITLE
Message queue length for a session does not change

### DISCUSSION THREAD

**Discussion by C-Aniruddh** [emqx/emqx#10844] [General]
I am running emqx in a k8s cluster, and I'm facing an issue where the connection keeps getting reset every few seconds. I tried testing it on golang paho client, MQTT X, and MQTT Explorer, but the behavior is same across all clients. 

Upon some debugging, I found this in client logs: 

```
2023-05-26T19:57:00+00:00 [SOCKET] testclient12@<ip>:60549 msg: emqx_connection_terminated, reason: {shutdown,message_queue_too_long}
```

For which I assumed that increasing the message queue length should resolve the issue. So I set the following in my environment config: 

```
EMQX_ZONE__INTERNAL__MAX_MQUEUE_LEN: 1000000
EMQX_ZONE__EXTERNAL__MAX_MQUEUE_LEN: 1000000
```

However, I still see it being set to 1000 in the dashboard (screenshot below)
<img width="719" alt="image" src="https://github.com/emqx/emqx/assets/6713004/4fe59cad-7e3b-4eb9-8ce7-0d5a47e99a3f">

I don't know if I am doing something wrong in this situation, or are there any other solutions to the connection termination.

**Comment by thalesmg** [comment]
Hi @C-Aniruddh .

Which EMQX version are you using?  From the screenshot, it seems to be `v5.0.x`, but the configuration you are trying to set are from `v4.4.x`.

To affect the maximum mqueue length you are seeing in the dashboard, assuming the client is connecting to the `default` zone, you need to set:

```yaml
EMQX_ZONES__DEFAULT__MQTT__MAX_MQUEUE_LEN: 1000000
```

![image](https://github.com/emqx/emqx/assets/16166434/0e02016a-75d4-4ef4-9eb5-b6ef88f1fbb2)

But the log you are seeing is not an alarm about the mqueue length (MQTT related), but actually the length of the Erlang process mailbox.  After this limit is reached, the connection process shuts down.  So you'd need to observe a client before it is shut down to potentially observe if that also reflects in an increase of the mqueue length.  This limit is controlled by the [`mailbox_max_size`](https://www.emqx.io/docs/en/v5.0/configuration/configuration-manual.html#force-shutdown) config.

The fact that the connection process is accumulating too many messages suggests that it may not be keeping up with the message rate.  A possible reason could be network congestion.

Are those clients subscribing to many topics, and possibly receiving many QoS 0 messages without any back-pressure?  Are they publishing messages as well?

**Reply by thalesmg** [reply]
Just to complement the previous message: if you want to try and increase the `mailbox_max_size`, I'd recommend at this moment setting it at the global level (`EMQX_FORCE_SHUTDOWN__MAILBOX_MAX_SIZE: xxx`) to test that out.

**Reply by m2b** [reply]
I know this topic is closed but where exactly does one do what your recommend?

**Comment by C-Aniruddh** [comment]
@thalesmg, adjusting the `FORCE_SHUTDOWN` parameters seems to resolve the error! Thank you for your help!
